airflow/providers/amazon/aws/operators/comprehend.py

```diff
@@ -0,0 +1,195 @@
+from airflow.utils.timezone import utcnow
+
+if TYPE_CHECKING:
+    import boto3
+
+    from airflow.utils.context import Context
+
+
+class ComprehendBaseOperator(AwsBaseOperator[ComprehendHook]):
+    """
+    This is the base operator for Comprehend Service operators (not supposed to be used directly in DAGs).
+
+    :param input_data_config: The input properties for a PII entities detection job. (templated)
+    :param output_data_config: Provides `conﬁguration` parameters for the output of PII entity detection
+        jobs. (templated)
+    :param mode: Specifies whether the output provides the locations (offsets) of PII  entities or a file in
+        which PII entities are redacted. If you set the mode parameter to ONLY_REDACTION. In that case you
+        must provide a RedactionConfig.
+    :param data_access_role_arn: The Amazon Resource Name (ARN) of the IAM role that grants Amazon Comprehend
+        read access to your input data. (templated)
+    :param language_code: The language of the input documents. (templated)
+    """
+
+    aws_hook_class = ComprehendHook
+
+    template_fields: Sequence[str] = aws_template_fields(
+        "input_data_config", "output_data_config", "data_access_role_arn", "language_code"
+    )
+
+    template_fields_renderers: dict = {"input_data_config": "json", "output_data_config": "json"}
+
+    def __init__(
+        self,
+        input_data_config: dict,
+        output_data_config: dict,
+        data_access_role_arn: str,
+        language_code: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_data_config = input_data_config
+        self.output_data_config = output_data_config
+        self.data_access_role_arn = data_access_role_arn
+        self.language_code = language_code
+
+    @cached_property
+    def client(self) -> boto3.client:
+        """Create and return the Comprehend client."""
+        return self.hook.conn
+
+    def execute(self, context: Context):
+        """Must overwrite in child classes."""
+        raise NotImplementedError("Please implement execute() in subclass")
+
+
+class ComprehendStartPiiEntitiesDetectionJobOperator(ComprehendBaseOperator):
+    """
+    Create a comprehend pii entities detection job for a collection of documents.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:ComprehendStartPiiEntitiesDetectionJobOperator`
+
+    :param input_data_config: The input properties for a PII entities detection job. (templated)
+    :param output_data_config: Provides `conﬁguration` parameters for the output of PII entity detection
+        jobs. (templated)
+    :param mode: Specifies whether the output provides the locations (offsets) of PII  entities or a file in
+        which PII entities are redacted. If you set the mode parameter to ONLY_REDACTION. In that case you
+        must provide a RedactionConfig.
```

Comment:
Provided in kwargs?

Reply:
Yes it comes from pii entities kwargs , have updated little more description now 😄 .